lib/client/keyagent.go

```diff
@@ -71,8 +72,8 @@ type LocalKeyAgent struct {
 	// insecure allows to accept public host keys.
 	insecure bool
 
-	// siteName specifies site to execute operation.
-	siteName string
+	// sites specifies sites to execute operation.
+	sites []string
```

Comment:
Tbh I find the fact that we use this field to indicate which CAs we want to load quite confusing. Like, what is "sites to execute operation"?

Why not keep the old field here and instead make a `loadAllCAs bool` field which would get initialized wherever this agent is initialized from the same property you propagate everywhere. If it's "true", then this agent will load all CAs when checking host signature. You then probably wouldn't need the new `GetClusterNames()` method you added as well, and would be able to just use `GetTrustedCertsPEM()` depending on whether the `loadAllCAs` is set.

Wdyt?

Reply:
I'd prefer to keep `GetClusterNames()` so we don't have to change `checkHostCertificateForClusters()`, but otherwise sounds good.